### PR TITLE
Tests setup

### PR DESCRIPTION
**Pull request by mtchavez**
## Updates
* travis config: Use new architecture with `sudo: false`
* travis config: Add go 1.5 to build matrix
* Update Makefile
  * `make deps` – builds dependencies
  * `make alldeps` – builds dependencies of package and tests
  * `make updatedeps` – will install package dependencies with update flag
  * `make test` – Runs tests with `go vet`
  * `make ci` – Runs tests for ci environment
  * `make bench` – Runs benchmark tests
* Add badges to `README`

## Vet issues

* fixed: `examples/appengine/hello.go:31: unreachable code`
* fixed: `possible formatting directive in log call`
* Issues with `tip` version of golang

  ```
  bugsnag_test.go:341: ExampleAutoNotify should return nothing
  bugsnag_test.go:349: ExampleRecover should be niladic
  bugsnag_test.go:407: ExampleNotify_details should be niladic
  errors/error_test.go:78: ExampleErrorf should be niladic
  errors/error_test.go:78: ExampleErrorf should return nothing
  errors/error_test.go:85: ExampleNewError should return nothing
  errors/error_test.go:85: ExampleNewError refers to unknown identifier: NewError
  errors/error_test.go:90: ExampleNewError_skip refers to unknown identifier: NewError
  errors/error_test.go:99: ExampleError_Stack should be niladic
  ```

### CODE REVIEW

File: examples/appengine/hello.go [hunk 1]
```diff
@@ -2,10 +2,10 @@ package mellow
 
 import (
 	"fmt"
-	"github.com/bugsnag/bugsnag-go"
-	"github.com/bugsnag/bugsnag-go/errors"
 	"net/http"
 	"os"
+
+	"github.com/bugsnag/bugsnag-go"
```
Comment (kattrali) [inline examples/appengine/hello.go]: `gofmt` will probably alphabetize this.

Comment (mtchavez) [inline examples/appengine/hello.go]: Yeah I have seen community discussion on the package format issue in the past. I think `go fmt` itself will not touch these. My go-fmt separates stdlib from 3rd party packages. Most open source stuff I have seen tends to follow this pattern as well, including libraries like packer, etcd, and kubernetes. ¯\(°_o)/¯

Comment (kattrali) [inline examples/appengine/hello.go]: Oh neat, I definitely agree that separating stdlib from third-party libs is better, though I was under the impression that `gofmt` didn't have an option for this. How did you make it configurable? (0_ō)

Comment (mtchavez) [inline examples/appengine/hello.go]: I use this with my editor https://godoc.org/golang.org/x/tools/cmd/goimports. I did a quick verify that `go fmt` didn't change the import order though, using `go1.5.1`

Comment (kattrali) [inline examples/appengine/hello.go]: Sounds good then. :+1: I wonder if the build will ever run...

**Comment by kattrali** [comment]
I ran the example apps suite manually, everything looks good here. :+1:

**Pull request merged**